### PR TITLE
add custom elements and 

### PR DESCRIPTION
**Pull request by meinradandermatt**
# Description
Add the possibility to toggle Custom elements and to detect the wrapping elements by tagName and Attributes. On this way a span can be used twice with other attributes.

# Changelog
- toggleTag checks on the tag and attributes
- on a selection a custom element can be toggled

### CODE REVIEW

File: spec/content.spec.js [hunk 1]
```diff
@@ -169,6 +169,38 @@ describe('Content', () => {
     })
   })
 
+  describe('getTagsByNameAndAttributes()', () => {
```
Comment (peyerluk) [inline spec/content.spec.js:172]: Please always use `function () {}` instead of `() => {` when using mocha.

File: spec/content.spec.js [hunk 2]
```diff
@@ -169,6 +169,38 @@ describe('Content', () => {
...
+    let range
+    beforeEach(() => {
+      range = rangy.createRange()
```
Comment (peyerluk) [inline spec/content.spec.js:175]: And here you can use `this.range = ...`. That's the default of how to set a variable with mocha.

File: spec/content.spec.js (hunk 1 above)
Comment (peyerluk) [reply spec/content.spec.js:172]: ... I see it's done like this in the whole file, but it is still wrong as it error prone and doesnt properly isolate tests when `() => {}` is used together with `this`. We should quickly fix the whole file.

File: src/content.js [hunk 3]
```diff
@@ -180,6 +180,24 @@ export function getTagsByName (host, range, tagName) {
...
+  return getTags(host, range, (node) => {
+    return node.nodeName === elem.nodeName.toUpperCase() &&
+      areSameAttributes(node.attributes, elem.attributes)
+  })
+}
+
+export function areSameAttributes (attrs1, attrs2) {
+  if (attrs1.length !== attrs2.length) return false
+
+  for (var i = 0; i < attrs1.length; i++) {
```
Comment (peyerluk) [inline src/content.js:193]: `var` -> `let`

File: src/content.js [hunk 4]
```diff
@@ -180,6 +180,24 @@ export function getTagsByName (host, range, tagName) {
   })
 }
 
+export function getTagsByNameAndAttributes (host, range, elem) {
+  return getTags(host, range, (node) => {
+    return node.nodeName === elem.nodeName.toUpperCase() &&
```
Comment (peyerluk) [inline src/content.js:185]: why is there only one `toUpperCase()?  in here?

File: src/content.js [hunk 5]
```diff
@@ -301,6 +325,18 @@ export function nuke (host, range, tagName) {
   })
 }
 
+// Unwrap all tags this range is affected by.
+// Can also affect content outside of the range.
+export function nukeElem (host, range, node) {
+  getTags(host, range).forEach((elem) => {
+    if (elem.nodeName !== 'BR' && (!node ||
+        (elem.nodeName === node.nodeName.toUpperCase() &&
```
Comment (peyerluk) [inline src/content.js:333]: same here: why is `toUpperCase()` only used once (this also goes for the line directly above: why not there?)

Comment (meinradandermatt) [reply src/content.js]: because I just adapted the nuke () function. But I changed it.

**Pull request merged**